xonsh/ptk_shell/updator.py

```diff
@@ -157,37 +157,53 @@ def submit_section(
 class PromptUpdator:
     """Handle updating multiple AsyncPrompt instances prompt/rprompt/bottom_toolbar"""
 
-    def __init__(self, session: PromptSession):
+    def __init__(self, shell):
+        from xonsh.ptk_shell.shell import PromptToolkitShell
+
         self.prompts: tp.Dict[str, AsyncPrompt] = {}
-        self.prompter = session
+        self.shell: PromptToolkitShell = shell
         self.executor = Executor()
+        self.futures = {}
+        self.attrs_loaded = None
 
     def add(self, prompt_name: tp.Optional[str]) -> tp.Optional[AsyncPrompt]:
         # clear out old futures from the same prompt
         if prompt_name is None:
             return None
 
-        if prompt_name in self.prompts:
```

Comment:
Why remove this check?

Reply:
the stop method already checks for this. I just removed the redundant check

Reply:
> I see it checks for `self.futures` but not `self.prompts`

Reply:
there are two checks in them 
![image](https://user-images.githubusercontent.com/6702219/130821721-afa52ae7-d77d-41be-bc51-1808fca015b5.png)